File: rustbox/src/tests/languages/test_files/lang_python/test_5_mle.py
import sys

def main():
    size = 1
    try:
        while True:
            arr = [0] * size
            size *= 2
            if size <= 0:
                break
    except MemoryError:
        print(f"Memory allocation failed at size = {size}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: rustbox/src/tests/languages/test_files/lang_cpp/test_5_mle.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    vector<vector<int>> data;
    int size = 1;
    
    try {
        while (true) {
            vector<int> arr(size, 0);
            data.push_back(arr);
            size *= 2;
            if (size <= 0) break;
        }
    } catch (const bad_alloc& e) {
        cout << "Memory allocation failed" << endl;
    } catch (const exception& e) {
        cout << "Error: " << e.what() << endl;
    }
    
    return 0;
}